lib/mapObjects/CRewardableObject.cpp

```diff
@@ -50,6 +50,38 @@ void CRewardableObject::selectRewardWthMessage(const CGHeroInstance * contextHer
 	sd.text = dialog;
 	sd.components = loadComponents(contextHero, rewardIndices);
 	cb->showBlockingDialog(&sd);
+
+}
+
+void CRewardableObject::grantAllRewardsWthMessage(const CGHeroInstance * contextHero, const std::vector<ui32> & rewardIndices, bool markAsVisit) const
```

Comment:
This is copy-pasted method `CRewardableObject::grantRewardWithMessage` with 1-2 different lines? Perhaps unify code?

Reply:
Yep, I think I can just call the existing  function for every reward in the loop.